src/backend/aspen/app/views/tests/test_auspice_redirect.py

```diff
@@ -1,33 +1,63 @@
 import json
 
+import requests
+
+from aspen.database.models import CanSee, DataType
 from aspen.test_infra.models.phylo_tree import phylorun_factory, phylotree_factory
 from aspen.test_infra.models.sample import sample_factory
 from aspen.test_infra.models.usergroup import group_factory, user_factory
 
 
 def test_auspice_redirect_view(session, app, client, mock_s3_resource, test_data_dir):
-    group = group_factory()
-    session.add(group)
-    session.commit()
+    viewer_group = group_factory()
+    can_see_group = group_factory("can_see")
+    wrong_can_see_group = group_factory("wrong_can_see")
+    no_can_see_group = group_factory("no_can_see")
+    can_see_group.can_be_seen_by.append(
+        CanSee(
+            viewer_group=viewer_group,
+            owner_group=can_see_group,
+            data_type=DataType.PRIVATE_IDENTIFIERS,
+        )
+    )
+    wrong_can_see_group.can_be_seen_by.append(
+        CanSee(
+            viewer_group=viewer_group,
+            owner_group=wrong_can_see_group,
+            data_type=DataType.SEQUENCES,
+        )
+    )
+    session.add_all(
+        [viewer_group, can_see_group, wrong_can_see_group, no_can_see_group]
+    )
 
-    user = user_factory(group)
+    user = user_factory(viewer_group)
 
-    sample_1 = sample_factory(
-        group,
-        private_identifier="private_identifer_1",
+    local_sample = sample_factory(
+        viewer_group,
+        private_identifier="private_identifier_1",
         public_identifier="public_identifier_1",
     )
-    sample_2 = sample_factory(
-        group,
-        private_identifier="private_identifer_2",
+    can_see_sample = sample_factory(
+        can_see_group,
+        private_identifier="private_identifier_2",
         public_identifier="public_identifier_2",
     )
+    wrong_can_see_sample = sample_factory(
+        wrong_can_see_group,
+        private_identifier="private_identifer_3",
+        public_identifier="public_identifier_3",
+    )
+    no_can_see_sample = sample_factory(
+        no_can_see_group,
+        private_identifier="private_identifer_4",
+        public_identifier="public_identifier_4",
+    )
```

Comment:
i'm a little confused on the difference between `no_can_see` and `wrong_can_see`, like when would a `wrong_can_see` happen in real life?

Reply:
`wrong_can_see` means like you can see trees of another group, but not the private identifiers.